src/app/auth/resetPassword/index.tsx

```diff
@@ -0,0 +1,177 @@
+  const checkPassword = (text: string) => {
+    setPassword(text);
+    if (text !== '') {
+      setHasUppercase(text !== text.toLowerCase());
+      setHasLowercase(text !== text.toUpperCase());
+      setHasNumber(/[0-9]/.test(text));
+      setHasLength(text.length >= 8);
+      //need to check that it is different from old password
+    }
+  };
+
+  const checkPasswordMatchConfirmPassword = (text: string) => {
+    setPassword(text);
+    if (text !== '') {
+      setIsMatching(text == confirmPassword);
+    }
+  };
+
+  const checkConfirmPasswordMatchPassword = (text: string) => {
+    setConfirmPassword(text);
+    if (text !== '') {
+      setIsMatching(text == password);
+    }
+  };
+
+  const changePassword = async () => {
+    setLoading(true);
+    const { error } = await updateUser({ password });
+
+    if (error) {
+      Alert.alert('Updating password failed');
+    } else {
+      await signOut();
+      router.replace('/auth/login');
+    }
+
+    setLoading(false);
+  };
+
+  return (
+    <View style={styles.container}>
+      <View>
+        <Text style={[globalStyles.h1]}>Create a New Password</Text>
+        <View style={styles.newPassword}>
+          <UserStringInput
+            placeholder="Password"
+            label="New password"
+            placeholderTextColor={colors.darkGrey}
+            onChange={text => {
+              setPassword(text);
+              checkPassword(text);
+              checkPasswordMatchConfirmPassword(text);
+            }}
+            value={password}
+            attributes={{
+              textContentType: 'password',
+              secureTextEntry: passwordTextHidden,
+            }}
+          >
+            <RNEIcon
+              name={passwordTextHidden ? 'visibility-off' : 'visibility'}
+              type="material"
+              style={styles.icon}
+              onPress={() => setPasswordTextHidden(!passwordTextHidden)}
+            />
+          </UserStringInput>
+        </View>
+
+        {password !== '' && (
+          <PasswordComplexityText
+            condition={hasUppercase}
+            message="At least 1 uppercase letter"
+          />
+        )}
+        {password !== '' && (
+          <PasswordComplexityText
+            condition={hasLowercase}
+            message="At least 1 lowercase letter"
+          />
+        )}
+        {password !== '' && (
+          <PasswordComplexityText
+            condition={hasNumber}
+            message="At least 1 number"
+          />
+        )}
+        {password !== '' && (
+          <PasswordComplexityText
+            condition={hasLength}
+            message="At least 8 characters"
+          />
+        )}
+
+        {/* functionality for this has not been implemented */}
+        {password !== '' && (
+          <PasswordComplexityText
+            condition={isDifferent}
+            message="Must be different than your old password"
+          />
+        )}
+
+        <View style={styles.newPassword}>
```

Comment:
According to the designs, the confirm password text box should not show up until all the conditions for the password are met. But, if the user edits their original password and the requirements are not met anymore, the text box should not go away.
I think the best way to do this is just by creating a new boolean state variable that's set true when all the requirements are true for the first time, and never becomes false. Then, you can `&&` that boolean with this `UserStringInput`